python/serverless-backend/README.md

```diff
@@ -0,0 +1,78 @@
+
+
+
+This project also demonstrates how to build a serverless backend application that will allow users to make authenticated API calls with API Gateway and Cognito and have the request be processed by a Lambda. Will use S3 to store images/files and DynamoDB to store any metadata needed. 
+
+# Stack will delpoy the following services: 
+- Amazon API Gateway - Entry point for your backend
+- AWS Lambda - Process JSON POST request
+- Amazon S3 - Bucket to store images or files 
+- Amazon DynamoDB _ Table for to store metadata
+- Amazon Cognito - Secure your backend with User Pool 
+- AWS IAM - Creates permissions 
+
+
+# Prerequisites:
+
+* An active AWS account
+* AWS Command Line Interface (AWS CLI) (Install and configure) 
+* AWS CDK Toolkit, (Install and configure)
+* Postman account or similar tool
+
+
+There are no manual steps involved in deploying this stack, however, in this example, Lambda code expects:
+1. JSON POST request must have the following keys: 
+
+{   userid : "string"
+    photo: "string - base64 format"
+    }
+2. Requires authenticated API calls. This solution assumes you have a frontend with a login UI, which allows you to configure the deployed Cognito with. If you do not have a UI and would like to test the API invoke URL via the CLI, you can follow the following instructions: 
+
+            - Create Cognito User 
+
+            Once the CloudFormation template is finished deploying, 
+            1. Navigate to CloudFormation in the console
+            2. Select the deployed stack and navigate to the “Resources” tab
+            3. Find the created User Pool by searching AWS::Cognito::UserPool and 4. click on Physical ID for UserPool
+            5. Navigate to “Users” and Click “create user”
+            6. Complete creating the user and save the username and password for later use
+            7. Go to your terminal to run some AWS CLI commands. 
+
+                The password for the new user needs to be permanent. To do so, run the following command (Change the value of user-pool-id to the deployed Cognito User Pool ID. Change the username and password to your created values):
+                aws cognito-idp admin-set-user-password \
+                --user-pool-id us-east-1_xxxxxx\
+                --username usernameExample \
+                --password passwordExample \
+                --permanent
+                Next, the IdToken of the account needs to be retrieved. To do so, run the following command (Change the value of client-id to the deployed Cognito User Pool Client ID. Change the username and password to your created values):
+                aws cognito-idp initiate-auth —region us-east-1 —auth-flow USER_PASSWORD_AUTH —client-id XXXXXXXXXXXXXXXXXX —auth-parameters USERNAME=usernameExample,PASSWORD=passwordExample
+                This command will return multiple keys. Save the IdToken for later use
+            - Use Postman to test your API endpoint
```

Comment:
Nitpick: a cURL call here would potentially be more useful.

Reply:
Using Postman here because it's easier to send image in a base64 format.